delfin/alert_manager/snmp_validator.py

```diff
@@ -153,31 +160,42 @@ def validate_connectivity(alert_source):
                   "reason: %s." % msg)
         raise exception.SNMPConnectionFailed(msg)
 
-    def _handle_validation_error(self, ctxt, storage_id):
+    def _handle_validation_result(self, ctxt, storage_id,
+                                  category=constants.Category.FAULT):
         try:
             storage = db.storage_get(ctxt, storage_id)
-            alert = {
-                'storage_id': storage['id'],
-                'storage_name': storage['name'],
-                'vendor': storage['vendor'],
-                'model': storage['model'],
-                'serial_number': storage['serial_number'],
-                'alert_id': constants.INTERNAL_ALERT_ID,
-                'sequence_number': 0,
-                'alert_name': 'SNMP connect failed',
-                'category': constants.Category.FAULT,
-                'severity': constants.Severity.MAJOR,
-                'type': constants.EventType.COMMUNICATIONS_ALARM,
-                'location': 'NetworkEntity=%s' % storage['name'],
-                'description': "SNMP connection to the storage failed. "
-                               "SNMP traps from storage will not be received.",
-                'recovery_advice': "1. The network connection is abnormal. "
-                                   "2. SNMP authentication parameters "
-                                   "are invalid.",
-                'occur_time': int(datetime.utcnow().timestamp()) * 1000,
-            }
-            self.exporter.dispatch(ctxt, alert)
+            serial_number = storage.get('serial_number')
+            if category == constants.Category.FAULT:
+                self.snmp_error_flag[serial_number] = True
+                self._dispatch_snmp_alert(ctxt, storage, category)
+            elif self.snmp_error_flag.get(serial_number, True):
+                self.snmp_error_flag[serial_number] = False
+                self._dispatch_snmp_alert(ctxt, storage, category)
         except Exception as e:
             msg = six.text_type(e)
             LOG.error("Exception occurred when handling validation "
                       "error: %s ." % msg)
+
+    def _dispatch_snmp_alert(self, ctxt, storage, category):
```

Comment:
function name can be changed!, This is not SNMP alert, 'dispatch_snmp_validation_alert" ?

Reply:
done